### PR TITLE
[terra-list] defect fixes

### PR DESCRIPTION
**Pull request by sugan2416**
### Summary

-  Updated styles to remove markers  and fix dividerstyles for list items 

Closes #

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-list/src/List.module.scss [hunk 1]
```diff
@@ -135,4 +135,8 @@
   .item-end {
     flex: 0 0 auto;
   }
+
+  .list-fill {
+    width: 100%;
```
Comment (supreethmr) [inline packages/terra-list/src/List.module.scss:140]: why do we need this ...??

Comment (sugan2416) [reply packages/terra-list/src/List.module.scss:140]: list under `listsection` needed to fill its immediate parent https://github.com/cerner/terra-core/pull/3749/files#diff-322ed14e3702ee33c7a170797460a1d83e655b49641aea7ad3636db21f3d5dc4R79

File: packages/terra-list/src/ListSection.jsx [hunk 2]
```diff
@@ -68,21 +68,19 @@ const ListSection = ({
 
   const theme = React.useContext(ThemeContext);
   const listClassNames = classNames(
-    cx(
-      'list',
-      theme.className,
-    ),
-    customProps.className,
```
Comment (supreethmr) [inline packages/terra-list/src/ListSection.jsx:75]: we would still need customProps.className

File: packages/terra-list/src/ListSectionHeader.module.scss [hunk 3]
```diff
@@ -86,4 +86,17 @@
...
+  .list-item {
+    align-items: center;
+    background-color: var(--terra-list-item-background-color);
+    display: flex;
+    justify-content: space-between;
+    margin: 0;
+    min-height: var(--terra-list-item-min-height);
+    padding: 0;
+    width: 100%;
+  }
```
Comment (supreethmr) [inline packages/terra-list/src/ListSectionHeader.module.scss:101]: this class is already defined on list.module.scss !!

File: packages/terra-list/src/ListSubsection.jsx [hunk 4]
```diff
@@ -65,21 +65,19 @@ const ListSubsection = ({
 
   const theme = React.useContext(ThemeContext);
   const listClassNames = classNames(
-    cx(
-      'list',
-      theme.className,
-    ),
-    customProps.className,
```
Comment (supreethmr) [inline packages/terra-list/src/ListSubsection.jsx:72]: same here : Does customProps.className removed purposefully

**Comment by supreethmr** [comment]
Can we WDIO test updated for list with dividers, section, section-header ??

File: packages/terra-list/src/ListSectionHeader.module.scss (hunk 3 above)
Comment (sugan2416) [reply packages/terra-list/src/ListSectionHeader.module.scss:101]: ListSectionHeader only imports ListSectionHeader.module.scss https://github.com/cerner/terra-core/blob/f19aef99b61648edb2b5bfa94028db8954ec9c48/packages/terra-list/src/ListSectionHeader.jsx#L7

File: packages/terra-list/src/ListSectionHeader.jsx [hunk 5]
```diff
@@ -112,9 +112,9 @@ const ListSectionHeader = ({
   }
 
   return (
-    <li className={cx(theme.className)}>
+    <li {...customProps} className={cx('list-item', theme.className)}>
       <Element className={cx('title')}>
-        <div {...customProps} {...attrSpread} className={sectionHeaderClassNames} ref={refCallback}>
+        <div {...attrSpread} className={sectionHeaderClassNames} ref={refCallback}>
```
[the remark this answers is not in the document]
Comment (sugan2416) [reply packages/terra-list/src/ListSectionHeader.jsx:117]: yes. Previously li itself was set to role=heading due to which `sectionHeaderClassNames` was applied to it. 
Now the li wraps section header

**Comment by sugan2416** [comment]
> Can we WDIO test updated for list with dividers, section, section-header ??

Added in https://github.com/cerner/terra-core/pull/3749/commits/17a1f59ee7172a9ed77445627718d8f6d9de7e4d

**Review by supreethmr — APPROVED** [review]

**Comment by kschuste** [comment]
@sugan2416  @supreethmr 
do you know when this will get released?

**Pull request merged**